src/ceph.c

```diff
@@ -522,18 +517,24 @@ static size_t count_parts(char const *key) {
  */
 static int parse_keys(char *buffer, size_t buffer_size, const char *key_str) {
   char tmp[2 * buffer_size];
+  size_t tmp_size = sizeof(tmp);
 
   if (buffer == NULL || buffer_size == 0 || key_str == NULL ||
       strlen(key_str) == 0)
     return EINVAL;
-
-  if ((count_parts(key_str) > 2) && has_suffix(key_str, ".type")) {
-    /* strip ".type" suffix iff the key has more than two parts. */
-    size_t sz = strlen(key_str) - strlen(".type") + 1;
-
-    if (sz > sizeof(tmp))
-      sz = sizeof(tmp);
-    sstrncpy(tmp, key_str, sz);
+  /* Strip suffix if it is ".type" or one of latency metric suffix. */
+  if (count_parts(key_str) > 2) {
+    if (has_suffix(key_str, ".type")) {
+      cut_suffix(tmp, tmp_size, key_str, ".type");
+    } else if (has_suffix(key_str, ".avgcount")) {
+      cut_suffix(tmp, tmp_size, key_str, ".avgcount");
+    } else if (has_suffix(key_str, ".sum")) {
+      cut_suffix(tmp, tmp_size, key_str, ".sum");
+    } else if (has_suffix(key_str, ".avgtime")) {
+      cut_suffix(tmp, tmp_size, key_str, ".avgtime");
+    } else {
+      sstrncpy(tmp, key_str, sizeof(tmp));
```

Comment:
You're using `tmp_size` above and `sizeof(tmp)` here. Please unify this. Since `tmp_size` is not modified, it's more common to just use `sizeof(tmp)` over and over again.